app/collections/products_collection.rb

```diff
@@ -25,7 +25,7 @@ def initialize(user: nil, shop: nil)
   #
   # I'm talk to discuss about this if is not clear : )
   def fetch
-    Product.all
+    Product.includes(:reviews).all
```

Comment:
Includes improve the query by telling Rails to do an SQL join with reviews when asking for the product. This way when doing `@product.reviews` this query is already done